pkg/clients/sls/index.go

```diff
@@ -104,8 +104,10 @@ func IsIndexNotFoundError(err error) bool {
 	if err == nil {
 		return false
 	}
-	if e, ok := errors.Cause(err).(*sdk.Error); ok && (e.Code == ErrCodeLogstoreIndexNotExist) {
-		return true
+	var slserr *sdk.Error
+	cause := errors.Cause(err)
```

Comment:
@jastang I think we may be able to avoid calling `errors.Cause` given that `errors.As` unwraps until it finds an error that matches in the chain (ref: https://pkg.go.dev/errors#As)